src/NuGet.Core/NuGet.Packaging/Licenses/LicenseExpressionToken.cs

```diff
@@ -0,0 +1,32 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+namespace NuGet.Packaging.Licenses
+{
+    /// <summary>
+    /// Represents a token of a parsed license expression. The tokens are either operators, parentheses or values. 
+    /// </summary>
+    internal class LicenseExpressionToken
+    {
+        /// <summary>
+        /// The token type
+        /// </summary>
+        internal LicenseTokenType TokenType { get; }
+
+        /// <summary>
+        /// The value of this token.
+        /// </summary>
+        internal string Value { get; }
+
+        internal LicenseExpressionToken(string value, LicenseTokenType tokenType)
+        {
+            Value = value;
```

Comment:
can this `value` be `null`? If not, change this to `  Value = value ?? throw new ArgumentNullException(nameof(value));`